### PR TITLE
Issue #30: core repository layer and schema contract docs

### PR DESCRIPTION
**Pull request by stranske**
> **Source:** Issue #30

#### Scope
Contract docs and repository abstractions are needed to keep schema usage consistent across services.

### Related Issues/PRs
- [#11](https://github.com/stranske/Inv-Man-Intake/issues/11)
- [#7](https://github.com/stranske/Inv-Man-Intake/issues/7)

#### Tasks
- [x] Implement repository methods for firm/fund/document create/read/update
- [x] Implement query helpers for version history and provenance lookups
- [x] Write contract documentation for core schema and usage examples
- [x] Add tests for repository behavior and edge-case queries
- [x] Add developer notes on extension strategy for optional fields

#### Acceptance criteria
- [x] Repository layer supports documented core access patterns
- [x] Contract docs match implemented schema and query behavior
- [x] Tests verify repository methods for expected and edge cases
- [x] Extension notes exist for adding new optional fields safely

### CODE REVIEW

File: docs/contracts/core_schema.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+This creates deterministic ordering for downstream processing.
+
+## Provenance Lookup Behavior
+
+`list_provenance_rows(document_id)`:
+- Returns `()` when `extracted_fields` table is absent.
+- Returns ordered `(field_key, value, source_page)` rows when present.
+
+This allows repository code to remain forward-compatible while provenance tables are rolled out incrementally.
```
[the remark this answers is not in the document]
Comment (stranske) [reply docs/contracts/core_schema.md]: Fixed in commit 0e6fdaf. Added a full **Extension Strategy** section at the end of the contract doc covering: (1) new columns must be nullable/defaulted, (2) renaming/removing columns is a breaking change requiring a version bump, (3) new tables should degrade gracefully when absent (following the `list_provenance_rows` pattern), and (4) API methods must preserve existing signatures. Also includes consumer-facing guarantees about optional field evolution and v1 stability.
